src/testdir/test_gettext_cp1251.vim

```diff
@@ -0,0 +1,13 @@
+source check.vim
+
+" Test for gettext()
+func Test_gettext()
+  if has('bind_codeset')
+    set encoding=cp1251
```

Comment:
please don't change the encoding, perhaps you need `:scriptencoding` instead?

Reply:
No, I really need to change the ``encoding``, because the string is to be sent to the terminal after this. I re-set the encoding in the test file, though...

Reply:
Hm, IIRC, setting encoding in the active vim session may invalidate all stored string data. So this is really a bad idea to change the encoding.

Reply:
This is 'just' a test file. In the wild, you wouldn't change ``encoding`` on the fly (I expect this to be set in the user's (or system-wide!) .vimrc). But, I need to see that my changes work, and I see no other way to test this...

Always using UTF-8 isn't going to work if the user's terminal works in cp1251 (or does it???), and both you and Bram were against specifying the encoding specifically, so changing the encoding is the only way to test this change.

I'm happy to be proven wrong, though, since this _is_ an 'ugly' bit of code, and hard to test without having a terminal (or language!) using these encodings.